doc.md

```diff
@@ -0,0 +1,110 @@
+```
+FILES_LOCATION=/orca_repo/etc
+PROJECT=hackseq17_6
+
+docker run -t --name $PROJECT -m 100GB --memory-reservation 20GB -v $FILES_LOCATION/hosts:/etc/hosts:ro -v /projects/hackathon/2017:/home/projects -v $FILES_LOCATION/passwd_hs17:/etc/passwd:ro -v $FILES_LOCATION/group_hs17:/etc/group:ro -w /home/projects/$PROJECT -v /home/userhome:/home/userhome -v /tmp/.X11-unix:/tmp/.X11-unix --net host bcgsc/orca-hackseq:2017.10.12
+```
+
+*Details*:
+
+`--name $PROJECT`: a name of the container = a name of the team 
+
+`-m 100GB`: The maximum amount of memory that a container can use.
+`--memory-reservation 20GB`: a soft limit, which is activated when Docker detects low memory or contention on the host machine. A server has 128 GB RAM / 5 teams = ~20 GB.
+
+`-v $FILES_LOCATION/hosts:/etc/hosts:ro`: a flag `–h` doesn’t work with a `--net host`, so, have to mount hosts file.
+
+`-v /projects/hackathon/2017:/home/projects`: mounts all project directories 
+
+`-v $FILES_LOCATION/passwd_hs17:/etc/passwd:ro -v $FILES_LOCATION/group_hs17:/etc/group:ro`: maps users of the container with users of the host machine
+
+`-w /home/projects/$PROJECT`:a work directory will be a project folder
+
+`-v /home/userhome:/home/userhome`: mounts home directories of all users
+
+`/tmp/.X11-unix:/tmp/.X11-unix --net host`: for running GUI application (X11)
+
+## Log in
+When a user logs in to the system the following script (`/usr/local/bin/orca.hackathon`) runs:
+
+```
+#!/bin/sh
+	 
+	if [ "$1" = "-c" ];then
+	        cmd=`echo $2 | grep "rsync\|scp"`
+	        if [ -z "$cmd" ]; then
+	                echo "======================================================="
+	                echo "           Welcome to BCGSC ORCA Service          "
+	                echo "Only direct rsync or scp file transfer commands allowed"
+	                echo "======================================================="
+	        else
+	            	exec $2
+	        fi
+	else
+	 
+	        USER=$(id -un)
+	        TEAM=$(id -gn)
+	 
+	        echo "Welcome to Hackseq 2017, USER $(id -un), Your home directory is mounted under /home/userhome/$USER"
+	        echo "Currently installed tools can be found via 'brew list' and '/home/linuxbrew/.linuxbrew/bin' in shell"
+	        echo "You can remotely transfer data via rsync/scp to your /home/userhome/$USER directory"
+	        echo "Your Hackathon project folder is mounted under /home/projects/$TEAM, please keep your raw data, scripts here"
+	   
+	        exec docker exec -it  -u $(id -u):$(id -g) $(id -gn) /usr/bin/env XAUTHORITY=/home/userhome/$USER/.Xauthority DISPLAY=${DISPLAY} /bin/bash -l
+
+```
+
+*Details*:
+* The name of the container = the name of the primary group = `$(id -gn)`
+
+* `XAUTHORITY=/home/userhome/$USER/.Xauthority`: provides a path to user’s `.Xauthority` file, otherwise it takes root’ `.Xauthority` file, since the container was created by root.
+* `DISPLAY=${DISPLAY}`: provides a new `$DISPLAY` variable of the terminal.
```

Comment:
Use `docker -e` rather than `env`

Reply:
# My machine

```
❯❯❯ docker --version
Docker version 17.03.0-ce, build 60ccb22
❯❯❯ docker exec --help
Usage:	docker exec [OPTIONS] CONTAINER COMMAND [ARG...]

Run a command in a running container

Options:
  -d, --detach               Detached mode: run command in the background
      --detach-keys string   Override the key sequence for detaching a container
  -e, --env list             Set environment variables (default [])
      --help                 Print usage
  -i, --interactive          Keep STDIN open even if not attached
      --privileged           Give extended privileges to the command
  -t, --tty                  Allocate a pseudo-TTY
  -u, --user string          Username or UID (format: <name|uid>[:<group|gid>])
```

# docker02

```
❯❯❯ /usr/bin/docker --version
Docker version 1.12.5, build 7392c3b
❯❯❯ /usr/bin/docker exec --help

Usage:	docker exec [OPTIONS] CONTAINER COMMAND [ARG...]

Run a command in a running container

  -d, --detach         Detached mode: run command in the background
  --detach-keys        Override the key sequence for detaching a container
  --help               Print usage
  -i, --interactive    Keep STDIN open even if not attached
  --privileged         Give extended privileges to the command
  -t, --tty            Allocate a pseudo-TTY
  -u, --user           Username or UID (format: <name|uid>[:<group|gid>])
```

Let's ask Brendan to upgrade the version of Docker that we're using. Please add it to the agenda for tomorrow's meeting.